### ISSUE TITLE
BlockTrades requires users to create an account to trade

### ISSUE THREAD

**Issue by abitmore** [bitshares/bitshares-ui#2695]
**Is your feature request related to a problem? Please describe.**
https://steemit.com/blocktrades/@blocktrades/blocktrades-users-must-now-create-an-account-to-trade-with-us

**Describe the solution you'd like**
Update UI.

**Describe alternatives you've considered**
Disable blocktrades.

**Additional context**

**Comment by dnotestein** [comment]
I've assigned a members of our team to update the BitShares UI for this. In the meantime, the bridge should continue to function, but trade sizes will be limited.

**Comment by startailcoon** [comment]
Awaiting PR from BlockTrades team.

**Comment by sschiessl-bcp** [comment]
@dnotestein 

Please present beforehand how you envision handling the user login from the UI side. Two options:
 - add a link to blocktrades signup, blocktrades registers the bitshares account 
 - user can signup directly in the UI, which would require KYC input in the UI

In both cases after signup, the UI for the user should be as before, possible showing a "registered" or somethign indicator in the bridge modal.

**Comment by startailcoon** [comment]
This would impact the "Buy" modal when using Block Trades, which is at the moment the only one we support. This modal would require some changes as @sschiessl-bcp mentions. 

When selecting a system for the buy order, it would need to support to view if a user is logged in or not, regardless of the case if the user can sign up within the UI or not. A user would still need a login option for the bridge within the UI.

- Probable impact
  - New modal for Signing in to external services, storing authentication keys (**required**)
  - New modal for Signup to external services, with KYC and everything else required for this. (optional)

**Comment by sschiessl-bcp** [comment]
> This would impact the "Buy" modal when using Block Trades, which is at the moment the only bridge we support. This modal would require some changes as @sschiessl-bcp mentions.
> 
> When selecting a system for the buy order, it would need to support to view if a user is logged in or not, regardless of the case if the user can sign up within the UI or not. A user would still need a login option for the bridge within the UI.
> 
> * Probable impact
>   
>   * New modal for Signing in to external services, storing authentication keys (**required**)
>   * New modal for Signup to external services, with KYC and everything else required for this. (optional)

Authentification keys can't / shouldn't be necessary. Gateway must / should recognize with the given bitshares account name if that account has KYC.

**Comment by startailcoon** [comment]
> Authentification keys can't / shouldn't be necessary. Gateway must / should recognize with the given bitshares account name if that account has KYC.

Right, I was viewing this as a login with username and password. If it just goes by bitshares account name and a checked KYC for that account name, then there's no need. Lets see the requirements from BlockTrades.

**Comment by dnotestein** [comment]
With the recent hack of btsmd accounts, I think it's worth bringing back up this issue for possible inclusion in an update.

At the very least, it's a new use-case for when it could be useful for protecting buyers. And save some of bitshares reputation when such hacks take place.

I've been asked by abit to write a BSIP for this, but unfortunately I don't have the time for that. I took a quick look at the BSIP repo, and it doesn't seem to be a small task. Still, if there is someone with the time (or say, getting paid) to do so, I think it's an idea worth implementing.

**Comment by abitmore** [comment]
I think @dnotestein replied to the wrong issue. The correct issue should be https://github.com/bitshares/bitshares-core/issues/1033

**Comment by startailcoon** [comment]
@dnotestein do you have any updates for us?

> 
> 
> @dnotestein
> 
> Please present beforehand how you envision handling the user login from the UI side. Two options:
> 
>     * add a link to blocktrades signup, blocktrades registers the bitshares account
> 
>     * user can signup directly in the UI, which would require KYC input in the UI
> 
> 
> In both cases after signup, the UI for the user should be as before, possible showing a "registered" or somethign indicator in the bridge modal.
